cpp/src/reductions/scan/scan_inclusive.cu

```diff
@@ -264,10 +264,21 @@ std::unique_ptr<column> scan_inclusive(
   // If the input is a structs column, we also need to push down nulls from the parent output column
   // into the children columns.
   if (input.type().id() == type_id::STRUCT && output->has_nulls()) {
-    for (size_type idx = 0; idx < output->num_children(); ++idx) {
-      structs::detail::superimpose_nulls(
-        output->view().null_mask(), output->null_count(), output->child(idx), stream, mr);
-    }
+    auto const num_rows   = output->size();
+    auto const null_count = output->null_count();
+    auto content          = output->release();
+
+    // Build new children columns.
+    const auto current_mask = reinterpret_cast<bitmask_type const*>(content.null_mask->data());
+    std::for_each(
+      content.children.begin(), content.children.end(), [current_mask, stream, mr](auto& child) {
+        child = structs::detail::superimpose_nulls(
+          current_mask, cudf::UNKNOWN_NULL_COUNT, std::move(child), stream, mr);
```

Comment:
```suggestion
          current_mask, null_count, std::move(child), stream, mr);
```
The `current_mask == content.null_mask` so is this is not `null_count`?
I'd like to minimize further reliance on `UNKNOWN_NULL_COUNT` if possible.

Reply:
Oh yes, `null_count` should be used. Thanks.